### PR TITLE
Fix GPU card in node details

### PR DESCRIPTION
**Pull request by zaelgohary**
### Changes

- Update GridNode type in proxy
- Replace cards arr with GPU arr from grid proxy
- Apply changes in Playground
- Remove unused log
- Removed repeated type & unnecessary code

image

image

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3126
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3476

### Tested Scenarios

- List GPU cards of a down node 250
- Check node details of an up node
- Check node details of a down node w no gpu

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by amiraabouhadid** [comment]
please add test cases

File: packages/gridproxy_client/src/modules/gateways.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export interface GPUCard {
   vendor: string;
   device: string;
   contract: number;
+  updated_at: number;
```
Comment (maayarosama) [inline packages/gridproxy_client/src/modules/gateways.ts:49]: why is this added?

**Comment by maayarosama** [comment]
Couldn't test because there's no standby nodes  with gpu cards on dev or main

**Comment by zaelgohary** [comment]
> Couldn't test because there's no standby nodes with gpu cards on dev or main

You can test with an offline node.

File: packages/gridproxy_client/src/modules/gateways.ts (hunk 1 above)
Comment (zaelgohary) [reply packages/gridproxy_client/src/modules/gateways.ts:49]: I replaced the cards array we add to GridNode with the gpu coming from grid proxy directly. Check https://gridproxy.dev.grid.tf/nodes/161/gpu

**Review by maayarosama — APPROVED** [review]

**Review by amiraabouhadid — COMMENTED** [review]
Is this [loadGpu](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/8d25b9a34a947c22ce867867138947dae82cd5ab/packages/gridproxy_client/src/modules/nodes.ts#L12) still necessary then?

**Review by amiraabouhadid — CHANGES_REQUESTED** [review]
Did not work for an offline node :(
image
image
Am I missing something? Gpu card does not load even after pressing try again.

**Comment by zaelgohary** [comment]
> Did not work for an offline node :( 

I made changes in Grid Proxy Client. Did you build before running the playground?

**Comment by amiraabouhadid** [comment]
> > Did not work for an offline node :(
> 
> I made changes in Grid Proxy Client. Did you build before running the playground?

my bad, was testing it on staging instead of locally, works
image

why is this [loadGpu](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/8d25b9a34a947c22ce867867138947dae82cd5ab/packages/gridproxy_client/src/modules/nodes.ts#L12) still necessary though?

**Comment by zaelgohary** [comment]
@amiraabouhadid I did some refactoring. You can test now.

**Review by amiraabouhadid — COMMENTED** [review]
I get this error when I press the x beside the selected GPU 
image
````
[Uncaught Error]
- Message: Cannot read properties of undefined (reading 'vendor')
- Stack: TypeError: Cannot read properties of undefined (reading 'vendor')
    at getNodeTwinDetailsCard (http://localhost:5173/src/components/node_details_cards/gpu_details_card.vue:43:37)
    at http://localhost:5173/src/components/node_details_cards/gpu_details_card.vue:23:25
    at callWithErrorHandling (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:1584:18)
    at callWithAsyncErrorHandling (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:1592:17)
    at job (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:3181:9)
    at callWithErrorHandling (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:1584:32)
    at flushJobs (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:1782:9)
- Constructor: TypeError
(anonymous) @ @sentry_vue.js?v=62295903:851
(anonymous) @ logger-interceptor.js?v=62295903:60
app.config.errorHandler @ main.ts:18
app.config.errorHandler @ @sentry_vue.js?v=62295903:22512
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
handleError @ chunk-S4KY57GZ.js?v=62295903:1625
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1586
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
job @ chunk-S4KY57GZ.js?v=62295903:3181
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
flushJobs @ chunk-S4KY57GZ.js?v=62295903:1782
Promise.then
queueFlush @ chunk-S4KY57GZ.js?v=62295903:1695
queueJob @ chunk-S4KY57GZ.js?v=62295903:1689
scheduler @ chunk-S4KY57GZ.js?v=62295903:3203
triggerEffect @ chunk-S4KY57GZ.js?v=62295903:633
triggerEffects @ chunk-S4KY57GZ.js?v=62295903:623
triggerRefValue @ chunk-S4KY57GZ.js?v=62295903:1230
set value @ chunk-S4KY57GZ.js?v=62295903:1274
set @ chunk-S4KY57GZ.js?v=62295903:1292
_createVNode.onUpdate:modelValue._cache.<computed>._cache.<computed> @ gpu_details_card.vue:27
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
emit @ chunk-S4KY57GZ.js?v=62295903:2093
set @ chunk-TJM6J5C4.js?v=62295903:124
set value @ chunk-S4KY57GZ.js?v=62295903:1407
onModelUpdate @ vuetify_components.js?v=62295903:9343
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
emit @ chunk-S4KY57GZ.js?v=62295903:2093
set @ chunk-TJM6J5C4.js?v=62295903:124
set value @ chunk-S4KY57GZ.js?v=62295903:1407
(anonymous) @ vuetify_components.js?v=62295903:8554
Promise.then
nextTick @ chunk-S4KY57GZ.js?v=62295903:1667
onClear @ vuetify_components.js?v=62295903:8553
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
invoker @ chunk-S4KY57GZ.js?v=62295903:9414
sentryWrapped @ @sentry_vue.js?v=62295903:7589
Show 36 more frames
Show lessUnderstand this error
gpu_details_card.vue:27 [Vue warn]: Error in watcher callback: "TypeError: Cannot read properties of undefined (reading 'vendor')"

found in

---> <GPUDetailsCard> at /home/amira/tfgrid-sdk-ts/packages/playground/src/components/node_details_cards/gpu_details_card.vue... (6 recursive calls)
       <VDialog>
         <NodeDetails> at /home/amira/tfgrid-sdk-ts/packages/playground/src/components/node_details.vue... (6 recursive calls)
           <DeploymentListManager> at /home/amira/tfgrid-sdk-ts/packages/playground/src/components/deployment_list_manager.vue
             <VMain>... (2 recursive calls)
               <App> at /home/amira/tfgrid-sdk-ts/packages/playground/src/App.vue
                 <AsyncComponentWrapper>
(anonymous) @ @sentry_vue.js?v=62295903:22521
consoleSandbox @ @sentry_vue.js?v=62295903:418
app.config.errorHandler @ @sentry_vue.js?v=62295903:22520
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
handleError @ chunk-S4KY57GZ.js?v=62295903:1625
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1586
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
job @ chunk-S4KY57GZ.js?v=62295903:3181
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
flushJobs @ chunk-S4KY57GZ.js?v=62295903:1782
Promise.then
queueFlush @ chunk-S4KY57GZ.js?v=62295903:1695
queueJob @ chunk-S4KY57GZ.js?v=62295903:1689
scheduler @ chunk-S4KY57GZ.js?v=62295903:3203
triggerEffect @ chunk-S4KY57GZ.js?v=62295903:633
triggerEffects @ chunk-S4KY57GZ.js?v=62295903:623
triggerRefValue @ chunk-S4KY57GZ.js?v=62295903:1230
set value @ chunk-S4KY57GZ.js?v=62295903:1274
set @ chunk-S4KY57GZ.js?v=62295903:1292
_createVNode.onUpdate:modelValue._cache.<computed>._cache.<computed> @ gpu_details_card.vue:27
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
emit @ chunk-S4KY57GZ.js?v=62295903:2093
set @ chunk-TJM6J5C4.js?v=62295903:124
set value @ chunk-S4KY57GZ.js?v=62295903:1407
onModelUpdate @ vuetify_components.js?v=62295903:9343
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
emit @ chunk-S4KY57GZ.js?v=62295903:2093
set @ chunk-TJM6J5C4.js?v=62295903:124
set value @ chunk-S4KY57GZ.js?v=62295903:1407
(anonymous) @ vuetify_components.js?v=62295903:8554
Promise.then
nextTick @ chunk-S4KY57GZ.js?v=62295903:1667
onClear @ vuetify_components.js?v=62295903:8553
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
invoker @ chunk-S4KY57GZ.js?v=62295903:9414
sentryWrapped @ @sentry_vue.js?v=62295903:7589
Show 36 more frames
Show lessUnderstand this error
main.ts:18 [Uncaught Error]
- Message: Cannot read properties of undefined (reading 'contract')
- Stack: TypeError: Cannot read properties of undefined (reading 'contract')
    at http://localhost:5173/src/components/node_details_cards/gpu_details_card.vue:103:46
    at Proxy.renderFnWithContext (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:2194:13)
    at Proxy.<anonymous> (http://localhost:5173/node_modules/.vite/deps/vuetify_components.js?v=62295903:17124:55)
    at renderComponentRoot (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:2243:17)
    at ReactiveEffect.componentUpdateFn [as fn] (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:7179:26)
    at ReactiveEffect.run (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:442:19)
    at instance.update (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:7229:52)
    at callWithErrorHandling (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:1584:32)
    at flushJobs (http://localhost:5173/node_modules/.vite/deps/chunk-S4KY57GZ.js?v=62295903:1782:9)
- Constructor: TypeError
(anonymous) @ @sentry_vue.js?v=62295903:851
(anonymous) @ logger-interceptor.js?v=62295903:60
app.config.errorHandler @ main.ts:18
app.config.errorHandler @ @sentry_vue.js?v=62295903:22512
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
handleError @ chunk-S4KY57GZ.js?v=62295903:1625
renderComponentRoot @ chunk-S4KY57GZ.js?v=62295903:2280
componentUpdateFn @ chunk-S4KY57GZ.js?v=62295903:7179
run @ chunk-S4KY57GZ.js?v=62295903:442
instance.update @ chunk-S4KY57GZ.js?v=62295903:7229
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
flushJobs @ chunk-S4KY57GZ.js?v=62295903:1782
Promise.then
queueFlush @ chunk-S4KY57GZ.js?v=62295903:1695
queueJob @ chunk-S4KY57GZ.js?v=62295903:1689
scheduler @ chunk-S4KY57GZ.js?v=62295903:3203
triggerEffect @ chunk-S4KY57GZ.js?v=62295903:633
triggerEffects @ chunk-S4KY57GZ.js?v=62295903:623
triggerRefValue @ chunk-S4KY57GZ.js?v=62295903:1230
set value @ chunk-S4KY57GZ.js?v=62295903:1274
set @ chunk-S4KY57GZ.js?v=62295903:1292
_createVNode.onUpdate:modelValue._cache.<computed>._cache.<computed> @ gpu_details_card.vue:27
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
emit @ chunk-S4KY57GZ.js?v=62295903:2093
set @ chunk-TJM6J5C4.js?v=62295903:124
set value @ chunk-S4KY57GZ.js?v=62295903:1407
onModelUpdate @ vuetify_components.js?v=62295903:9343
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
emit @ chunk-S4KY57GZ.js?v=62295903:2093
set @ chunk-TJM6J5C4.js?v=62295903:124
set value @ chunk-S4KY57GZ.js?v=62295903:1407
(anonymous) @ vuetify_components.js?v=62295903:8554
Promise.then
nextTick @ chunk-S4KY57GZ.js?v=62295903:1667
onClear @ vuetify_components.js?v=62295903:8553
callWithErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1584
callWithAsyncErrorHandling @ chunk-S4KY57GZ.js?v=62295903:1592
invoker @ chunk-S4KY57GZ.js?v=62295903:9414
sentryWrapped @ @sentry_vue.js?v=62295903:7589
Show 37 more frames
Show lessUnderstand this error
gpu_details_card.vue:27 [Vue warn]: Error in render function: "TypeError: Cannot read properties of undefined (reading 'contract')"

found in

---> <VCol>... (11 recursive calls)
       <VDialog>
         <NodeDetails> at /home/amira/tfgrid-sdk-ts/packages/playground/src/components/node_details.vue... (6 recursive calls)
           <DeploymentListManager> at /home/amira/tfgrid-sdk-ts/packages/playground/src/components/deployment_list_manager.vue
             <VMain>... (2 recursive calls)
               <App> at /home/amira/tfgrid-sdk-ts/packages/playground/src/App.vue
                 <AsyncComponentWrapper>
```

**Comment by zaelgohary** [comment]
> I get this error when I press the x beside the selected GPU

Did you run make build in root?

**Comment by zaelgohary** [comment]
@amiraabouhadid issue was found on staging. 

image

It should be fixed now.

**Review by amiraabouhadid — APPROVED** [review]
